File: backends/dotnet/src/Zappy.Domain/Accounts/Customer.cs
namespace Zappy.Domain;

public sealed class Customer
{
    private Customer()
    {
    }

    public Customer(string id, EmailAddress email, string name, string passwordHash, DateTimeOffset createdAt)
    {
        Id = id;
        Email = email;
        Name = name;
        PasswordHash = passwordHash;
        CreatedAt = createdAt;
    }

    public string Id { get; private set; } = null!;

    public EmailAddress Email { get; private set; } = null!;

    public string Name { get; private set; } = null!;

    public string PasswordHash { get; private set; } = null!;

    public DateTimeOffset CreatedAt { get; private set; }
}

File: backends/node/subgraphs/accounts/src/domain/customer.ts
import type { EmailAddress } from "@zappy/shared";

export const minimumPasswordLength = 12;

export const maximumPasswordLength = 128;

export type Customer = {
  readonly id: string;
  readonly email: EmailAddress;
  readonly name: string;
  readonly createdAt: string;
};

export type PasswordVerdict = "acceptable" | "PASSWORD_TOO_SHORT" | "PASSWORD_TOO_LONG";

export function judgePassword(password: string): PasswordVerdict {
  if (password.length < minimumPasswordLength) {
    return "PASSWORD_TOO_SHORT";
  }
  return password.length > maximumPasswordLength ? "PASSWORD_TOO_LONG" : "acceptable";
}
